File: hospital.client/src/types/DispenseResponse.ts
export interface DispenseResponse {
  id: number;
  prescriptionId: number;
  patientId: number;
  pharmacistId: number;
  dispenseStatus: number;
  totalAmount: number;
  paymentMethod?: string | null;
  notes?: string | null;
  state: number;
  createdAt: string;
  createdBy: number;
  updatedBy?: number | null;
  updatedAt?: string | null;
}

export interface DispenseRequest {
  id?: number | null;
  prescriptionId?: number | null;
  patientId?: number | null;
  pharmacistId?: number | null;
  dispenseStatus?: number | null;
  totalAmount?: number | null;
  paymentMethod?: string | null;
  notes?: string | null;
  state?: number | null;
  createdBy?: number | null;
  updatedBy?: number | null;
}

/** Métodos de pago aceptados según RN-GLOBAL-004 */
export const PAYMENT_METHODS = [
  { label: "Efectivo", value: "EFECTIVO" },
  { label: "Tarjeta de Crédito (Visa/Mastercard)", value: "TARJETA_CREDITO" },
  { label: "Tarjeta de Débito", value: "TARJETA_DEBITO" },
] as const;

export type PaymentMethodValue = (typeof PAYMENT_METHODS)[number]["value"];

File: Hospital.Server/Entities/Response/DispenseResponse.cs
namespace Hospital.Server.Entities.Response
{
    public class DispenseResponse
    {
        public long Id { get; set; }
        public long PrescriptionId { get; set; }
        public long PatientId { get; set; }
        public long PharmacistId { get; set; }
        public int DispenseStatus { get; set; }
        public decimal TotalAmount { get; set; }
        public string? Notes { get; set; }
        public int State { get; set; }
        public string CreatedAt { get; set; } = string.Empty;
        public long CreatedBy { get; set; }
        public long? UpdatedBy { get; set; }
        public string? UpdatedAt { get; set; }
    }
}
